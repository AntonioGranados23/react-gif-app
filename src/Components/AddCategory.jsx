import { useState } from "react"

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('');


    // Obtenemos el valor del input, que finalmente es lo que se escribe
    const onInputChange = (event) => {
        setInputValue(event.target.value);
    } 

    // Controlamos el formulario para evitar que se actualice la pagina al presionar enter
    const onSubmit = (event) => {
        event.preventDefault();

        // Validamos que debe almenos un caracter para poder agregarlo al arreglo de categorias
        if(inputValue.trim() <= 1) return;

        // Agregamos la nueva categoria escrita en el input al arreglo
        onNewCategory(inputValue.trim());


        // Limpiamos el input al agregar la nueva categoria
        setInputValue('')
        
    }

  return (
    <form onSubmit={(event) => onSubmit(event)}>
        <input type="text"
        placeholder="Buscar Gifs"
        autoComplete="off"
        value={inputValue}
        onChange={onInputChange}
         />
    </form>
  )
}
