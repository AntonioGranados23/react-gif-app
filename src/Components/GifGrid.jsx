import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs"


export const GifGrid = ({category}) => {
  const {images, isLoading} = useFetchGifs(category);  

  return (
    <>
        <h3>{category}</h3>
        {
          isLoading && (<h2>Cargando...</h2>)
        }
        <div className="card-grid">
          {
            // validamos si el arreglo de imagenes esta vacio para mostrar un mensaje
            images.length === 0 ? (
              <h5 style={{color: "red"}}>No se encontraron resultados para "{category}"</h5>
            ): (
              // Recorremos el arreglo de imagenes obtenidas
              images.map((image) => ( 
                <GifItem key={image.id} {...image} /> 
              ))
            )
          }
        </div>
    </>
  )
}
