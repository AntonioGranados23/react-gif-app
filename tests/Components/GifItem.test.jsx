import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/Components/GifItem"

describe('Pruebas en el componente GifItem', () => {
    const title = 'Eren';
    const url = 'https://www.shingeki-no-kioyin.com.co/';
    test('Debe hacer match con el snapshot', () => {
        const {container} = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con la url y el alt indicado', () => {
        render(<GifItem  title={title} url={url} />);
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)

        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    });

    test('Debe mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
});