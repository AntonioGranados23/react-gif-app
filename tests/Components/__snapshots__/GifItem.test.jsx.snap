// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Eren"
      src="https://www.shingeki-no-kioyin.com.co/"
    />
    <p>
      Eren
    </p>
  </div>
</div>
`;
